### ISSUE TITLE
"Hide Locked Backgrounds" is misaligned to the left instead of in the centre

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#12702]
At User Icon > Backgrounds, the "Hide Locked Backgrounds" switch is on the left of the modal. It should be in the centre:
![image](https://user-images.githubusercontent.com/1495809/97104216-d0c46180-16fd-11eb-9a74-feef9df0e3c4.png)

**Comment by himanshus852** [comment]
Hi 
I would like to work on this issue .Please allow me.

**Comment by Alys** [comment]
@himanshus852 Thank you! Please go ahead! Post here if you have questions or run into trouble.

The person who originally reported this, @Zarathustrik, has posted to the Report a Bug guild again with some extra details that they've just noticed:

_"there is some problem with this bug in " Hide locked backgrounds". I reloaded my page it appers again, I reloaded again it disappears, then I reloaded again and it appears again. There is something starnge with this floatint bug, Thank you"_

From my testing just now, it seems to depend on which page you were looking at before going to the backgrounds. Here's one way to reproduce the bug:
1. Go to the Tasks page.
1. Reload it with your browser's refresh / reload button.
1. Click on User Icon > Backgrounds.
1. Observe that "Hide locked backgrounds" is misaligned.
1. Go to the Tavern page.
1. Reload it with your browser's refresh / reload button.
1. Click on User Icon > Backgrounds again.
1. Observe that "Hide locked backgrounds" is correctly aligned.

There may well be other ways to reproduce it.

**Comment by thewilloftheshadow** [comment]
@Alys labels ;)
I was about to ask for this one and saw that someone had claimed it but it was still marked Help Wanted

**Comment by anshuman9999** [comment]
Can I work on this issue?

**Comment by ivan-suarez** [comment]
Hi, I saw this issue open and decided to give it a go, and [I think I have it fixed](https://github.com/ivan-suarez/habitica/commit/251dd210b1591f9ba148f635ef271e0226948458). 

I can proceed to pull request if it's okay.

**Comment by Alys** [comment]
@anshuman9999 @ivan-suarez This issue is labelled as "in progress", not "help wanted", so it's not currently available to be worked on. See "Ways to Help" in [Guidance for Blacksmiths](https://habitica.fandom.com/wiki/Guidance_for_Blacksmiths) for more details about selecting issues and the label system. :)

@himanshus852 How are you going with this? Do you still want to work on it?

**Comment by shanaqui** [comment]
@himanshus852 Since you haven't replied, we'll make this issue available for others to work on again.

@thewilloftheshadow Since you were the first other person to inquire, are you interested in taking this up?

**Comment by DanAmbitous** [comment]
Hey is this issue still open?

**Comment by thewilloftheshadow** [comment]
Hey @shanaqui! I've actually promised myself I wouldn't work on anymore frontend jobs right now because they always frustrate me and mess up my day. Sorry about that!

**Comment by yiqcheng98** [comment]
Hi! Can I take this one?

**Comment by SachJay** [comment]
I dont mind acting as a backup if this ticket frees up after @yiqcheng98

**Comment by tirlochanarora16** [comment]
Hey, can I work on this issue?

**Comment by paglias** [comment]
Sorry for the delay everyone, I'm assigning this issue to @yiqcheng98 if you still want it!

**Comment by tirlochanarora16** [comment]
If he doesn't accept can I take it then?

**Comment by paglias** [comment]
If @yiqcheng98 isn't interested I'll go in order of who posted before but since @SachJay already has an assigned issue that would be you :)

**Comment by tirlochanarora16** [comment]
Okay

**Comment by JohnJSal** [comment]
I've been playing around with when this bug occurs based on @Alys's comment above, and I think I've figured out exactly how to reproduce it. Maybe this will be helpful to whoever works on it.

I went to every subsection of the Habitica website (Tasks, Items, Equipment, Market, Quests, Party, etc.) all the way across the bar, except for the Help tab, because those subsections lead to different websites (except Report a Bug and Habitica Help).

There are some sections that *always* misalign the toggle, whether you refresh the browser or not. These are Tasks, Items, Equipment, Stable, and Market.

All other sections (Quests, Seasonal Shop, Time Travelers, Party, Tavern Chat, My Guilds, Discover Guilds, Group, My Challenges, and Discover Challenges) will correctly center-align the toggle if you refresh the browser.

However, moving from any page that center-aligns the toggle to any page that left-aligns it will always "reset" the alignment back to left-aligned. For example:

1. Go to Shops > Quests
2. Refresh the browser
3. Alignment is center-aligned
4. Go to Inventory > Items
5. Alignment is left-aligned (no refresh needed for this to occur)
6. Go to Shops > Quest (or any section that center-aligns)
7. Alignment is still left-aligned
8. Refresh the browser
9. Alignment is center-aligned

So basically, whenever you visit a page that left-aligns the toggle, all pages thereafter will also left-align it until you refresh. Once you refresh, all pages that center-align the toggle will center-align it *until* you visit one of the pages that left-aligns it.

Hopefully this makes sense and is somewhat useful! Good luck!

**Comment by shanaqui** [comment]
Hi @yiqcheng98, are you still working on this? If not, or if we don't hear back from you within a week, I'll mark it as help wanted again so the issue doesn't go totally stale. :)

**Comment by j-dover** [comment]
Hi, I would like to work on this issue if it is available.

**Comment by Alys** [comment]
@j-dover Thanks! I've marked this as in progress for you. Post here if you have any questions!

**Comment by Alys** [comment]
I've deleted some old comments from here that weren't directly related to the bug / fix to make the remaining comments easier to find.

**Comment by lizapritchard** [comment]
Hi @Alys I would like to work on this

**Comment by j-dover** [comment]
Hi @Alys , I was able to get the toggle centered by modifying the .backgroundFilterToggle CSS class in website/client/src/components/creatorIntro.vue. The toggle stayed in the center throughout recreating the above scenarios and by going to different pages. 

I tried running the test suite according to [Creating and Running Automated Tests](https://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica's_Website_and_API#Creating_and_Running_Automated_Tests) to make sure the website tests successfully. I ran into the following error after stopping `npm start` and using `killall node` from "If you are not using Docker: ", and then using `npm run mongo:dev` from "Running the Full Test Suite". 

```
> habitica@4.189.1 mongo:dev /home/jenny/habitica
> run-rs -v 4.2.8 -l ubuntu1804 --keep --dbpath mongodb-data --number 1 --quiet

Skipping purge
Running '/home/jenny/habitica/node_modules/run-rs/4.2.8/mongod' [ 27017 ]
Restarting replica set...
(node:21280) Warning: Accessing non-existent property 'count' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:21280) Warning: Accessing non-existent property 'findOne' of module exports inside circular dependency
(node:21280) Warning: Accessing non-existent property 'remove' of module exports inside circular dependency
(node:21280) Warning: Accessing non-existent property 'updateOne' of module exports inside circular dependency
Error: failed to start mongod with options [
  '--port=27017',
  '--dbpath=mongodb-data/27017',
  '--bind_ip=localhost',
  '--replSet=rs'
]
2021-04-09T16:26:27.901-0400 I  CONTROL  [main] Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'
2021-04-09T16:26:27.906-0400 W  ASIO     [main] No TransportLayer configured during NetworkInterface startup
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] MongoDB starting : pid=21288 port=27017 dbpath=mongodb-data/27017 64-bit host=jenny-VirtualBox
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] db version v4.2.8
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] git version: 43d25964249164d76d5e04dd6cf38f6111e21f5f
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] allocator: tcmalloc
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] modules: none
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] build environment:
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten]     distmod: ubuntu1804
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten]     distarch: x86_64
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten]     target_arch: x86_64
2021-04-09T16:26:27.907-0400 I  CONTROL  [initandlisten] options: { net: { bindIp: "localhost", port: 27017 }, replication: { replSet: "rs" }, storage: { dbPath: "mongodb-data/27017" } }
2021-04-09T16:26:27.907-0400 E  STORAGE  [initandlisten] Failed to set up listener: SocketException: Address already in use
2021-04-09T16:26:27.908-0400 I  CONTROL  [initandlisten] now exiting
2021-04-09T16:26:27.908-0400 I  CONTROL  [initandlisten] shutting down with code:48

    at ChildProcess.<anonymous> (/home/jenny/habitica/node_modules/run-rs/src/topologyManagerPatch.js:152:13)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1048:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)

```

I only changed the CSS. It worked when I tested during development. I haven't committed and pushed my changes yet. 
How can I restore the mongodb replica set?

**Comment by Alys** [comment]
@lizapritchard Thank you for offering but @j-dover had already claimed this one. :)

@j-dover Sometimes the tests don't run well on local installs, so I'm sure the problems you've found aren't related to your change. If you'd like to submit a pull request, you'd be welcome to!

So sorry to both of you for the delay in my reply!
